Library/Tracking/Segment.swift

```diff
@@ -7,54 +7,27 @@ public extension Analytics {
   /**
    Returns an `Analytics` instance, with Segment, using an `AnalyticsConfiguration`.
    */
-  static func configuredClient() -> Analytics {
-    // Due to this being constructed at the same time as the environment we're not able to refer to the
-    // mainBundle on the environment here. We probably should if we want to test this.
-    let writeKey = Bundle.main.isRelease
-      ? Secrets.Segment.production
-      : Secrets.Segment.staging
-
+  static func configuredClient(withWriteKey writeKey: String) -> Analytics {
     let configuration = AnalyticsConfiguration(writeKey: writeKey)
     configuration
       .trackApplicationLifecycleEvents = true
 
     // Braze is always configured but feature-flagged elsewhere.
+    // Data sent to Braze is feature-flagged by the enabling/disabling Segment.
     configuration.use(SEGAppboyIntegrationFactory.instance())
 
     Analytics.setup(with: configuration)
 
+    // Disabled when initialized and enabled by our feature-flag configuration.
+    Analytics.shared().disable()
+
     return Analytics.shared()
   }
 }
 
 /**
  The `TrackingClientType` and `IdentifyingTrackingClient` protocols
- allow us to create mocks to test these code paths. The protocol exposes functions that are named similarly
- but differently so that we can perform the check to see that tracking is enabled before calling any of the
- functions on the library itself so as to not unintentionally contribute to tracking data during debugging.
+ allow us to create mocks to test these code paths.
  */
-
-extension Analytics: IdentifyingTrackingClient {
-  public func identify(userId: String?, traits: [String: Any]?) {
-    guard AppEnvironment.current.environmentVariables.isTrackingEnabled,
```

Comment:
Weirdly no tests failed removing this so it looks like we weren't testing it anyway 🤦‍♂️